packages/ses/src/error/README.md

```diff
@@ -0,0 +1,84 @@
+# Logging Errors
```

Comment:
This is thorough!

Perhaps we can front-load some user-facing take-aways:

* Writing defensive programs under SES requires carefully considering what an error reveals to code positioned to catch those errors up the call chain.
* To that end, SES introduces an `assert` global with functions that that effect errors with annotations that will be hidden from callers, and tames the `Error` constructor such that it hides the `stack` to parent callers.
* SES tames the global `console` and grants it the ability to reveal error annotations and stacks to the actual console.
* Both `assert` and `console` are both powerful globals that SES does not implicitly carry into child compartments. When creating a child compartment, add `assert` to the compartment’s globals. Either add `console` too, or add a wrapper that annotates the console with a topic.
* SES hides annotations and stack traces by default. To reveal them, use a mechanism like `process.on("uncaughtException")` in Node.js to catch the error and log it back to the `console` tamed by `lockdown`.
* In the scope of the Agoric software ecosystem, this architecture will allow us to eventually introduce a more powerful distributed `console` that can meaningfully capture stack traces for a distributed debugger, based on the design of Causeway (linked).

We should have user-facing documentation for the usage of `assert` and `console`, including that last bit for revealing messages, either high in this README or in the SES README, until we have a better story for discovering SES documentation topics. My preference is the SES README so folks can use simple <kbd>⌘</kbd>+<kbd>F</kbd> to find relevant API docs.

Reply:
Good text. I took it directly with only minor edits.

The "Logging Errors" section near the end of the SES README is already findable by searching on `assert` or `console` and already links to `src/error/README.md`, so I put this high level summary there. However, we don't yet have user-level documentation. I extracted that to https://github.com/Agoric/SES-shim/issues/486 and going to resolve this in place.